pwndbg/heap/ptmalloc.py

```diff
@@ -144,12 +160,33 @@ def get_arena(self, arena_addr=None):
         return pwndbg.memory.poi(self.malloc_state, arena_addr)
 
 
-    def get_bounds(self):
+    def get_arena_for_chunk(self,addr):
+        chunk = pwndbg.memory.poi(self.malloc_state,addr)
+        _,_,nm = self.chunk_flags(chunk['size'])
+        if nm:
+            r=self.get_arena(arena_addr=self.get_heap(addr)['ar_ptr'])
+        else:
+            r=self.main_arena
+        return r
+            
+    def get_region(self,addr=None):
         """
-        Finds the heap bounds by using mp_ structure's sbrk_base property
+        Finds the memory region used for heap by using mp_ structure's sbrk_base property
         and falls back to using /proc/self/maps (vmmap) which can be wrong
         when .bss is very large
+        For non main-arena heaps use heap_info struct provieded at the beging of the page.
         """
+
+        if addr:
+            heap  = self.get_heap(addr)
+            base  = int(heap.address) + self.heap_info.sizeof + self.malloc_state.sizeof
+            page  = pwndbg.vmmap.find(base)
+            ## trim whole page to look exact like heap
```

Comment:
It might be better to define a `trim` function in `Page` which would:
* check if the trimming is correct, e.g. have an assertion against making the page bigger
* return completely new `Page` object, so the old one could be re-used if needed